scheduler/util.go

```diff
@@ -341,6 +341,15 @@ func tasksUpdated(jobA, jobB *structs.Job, taskGroup string) bool {
 	a := jobA.LookupTaskGroup(taskGroup)
 	b := jobB.LookupTaskGroup(taskGroup)
 
+	// Check Job level Affinities and Constraints
```

Comment:
I think a cleaner way to do this would be to merge all affinities into a single slide and then check for equality. (same with Constraints). 
something like https://github.com/hashicorp/nomad/blob/master/scheduler/rank.go#L592 but you'd build the slice here. 

Also, Spreads is another similar concept (spreading instances based on node metadata) where we;d want to do the same thing here (apply a destructive update if the values changed).

Reply:
This would address your question above (when something moves from a job to a task group level). It still wouldn't address cases where the ordering of multiple affinity stanza's changed but that seems like a rare edge case so an unnecessary destructive update is fine.

Reply:
Makes sense and I'll add a check for spreads as well, thanks